### PR TITLE
LIME-1007 - Adding alarms for concurrent execution thresholds

### PR DESCRIPTION
**Pull request by ElliotMurphyGDS**
### What changed

 - Added alarms that trigger when any lambda comes close to concurrent execution thresholds
 - Updated bad token alarm to only fire when 5 bad tokens have been received within a minute time period

### CODE REVIEW

File: infrastructure/lambda/template.yaml [hunk 1]
```diff
@@ -1002,6 +1002,121 @@ Resources:
...
+      MetricName: ConcurrentExecutions
+      Namespace: AWS/Lambda
+      Statistic: Average
+      Dimensions:
+        - Name: FunctionName
+          Value: !Ref IdentityCheckingFunction
+      Period: 60
+      EvaluationPeriods: 15
+      DatapointsToAlarm: 12
+      Threshold: 3
```
Comment (gds-jbentham) [inline infrastructure/lambda/template.yaml:1024]: I think this needs to be higher as 4 - 5  is average could we make this 9 which would align with our peak

File: infrastructure/lambda/template.yaml [hunk 2]
```diff
@@ -1002,6 +1002,121 @@ Resources:
...
+      MetricName: ConcurrentExecutions
+      Namespace: AWS/Lambda
+      Statistic: Average
+      Dimensions:
+        - Name: FunctionName
+          Value: !Ref IssueCredentialFunction
+      Period: 60
+      EvaluationPeriods: 15
+      DatapointsToAlarm: 12
+      Threshold: 3
```
Comment (gds-jbentham) [inline infrastructure/lambda/template.yaml:1047]: Can we make this 5 or 6 as 3 is the norm so dont want to report when not required

File: infrastructure/lambda/template.yaml [hunk 3]
```diff
@@ -1002,6 +1002,121 @@ Resources:
...
+      MetricName: ConcurrentExecutions
+      Namespace: AWS/Lambda
+      Statistic: Average
+      Dimensions:
+        - Name: FunctionName
+          Value: !Sub "${CommonStackName}-SessionFunction"
+      Period: 60
+      EvaluationPeriods: 15
+      DatapointsToAlarm: 12
+      Threshold: 3
```
Comment (gds-jbentham) [inline infrastructure/lambda/template.yaml:1070]: Same as above

File: infrastructure/lambda/template.yaml [hunk 4]
```diff
@@ -1002,6 +1002,121 @@ Resources:
...
+      MetricName: ConcurrentExecutions
+      Namespace: AWS/Lambda
+      Statistic: Average
+      Dimensions:
+        - Name: FunctionName
+          Value: !Sub "${CommonStackName}-AuthorizationFunction"
+      Period: 60
+      EvaluationPeriods: 15
+      DatapointsToAlarm: 12
+      Threshold: 3
```
Comment (gds-jbentham) [inline infrastructure/lambda/template.yaml:1093]: Same as above

File: infrastructure/lambda/template.yaml [hunk 5]
```diff
@@ -1002,6 +1002,121 @@ Resources:
...
+      MetricName: ConcurrentExecutions
+      Namespace: AWS/Lambda
+      Statistic: Average
+      Dimensions:
+        - Name: FunctionName
+          Value: !Sub "${CommonStackName}-AccessTokenFunction"
+      Period: 60
+      EvaluationPeriods: 15
+      DatapointsToAlarm: 12
+      Threshold: 3
```
Comment (gds-jbentham) [inline infrastructure/lambda/template.yaml:1116]: As above

**Review by smsgds — APPROVED** [review]

**Pull request merged**